### DISCUSSION TITLE
Support for the mode button on recent Nitro laptops

### DISCUSSION THREAD

**Discussion by overcrook** [JafarAkhondali/acer-predator-turbo-and-rgb-keyboard-linux-module#178] [Ideas]
Hello @JafarAkhondali , thanks for this repo and for useful videos!

I have a Nitro AN17-41, which has a "mode" button instead of "turbo" button. In Windows, this button doesn't just toggle Turbo mode, it actually loops through some predefined modes, such as:
* Eco
* Quiet
* Balanced
* Performance
* Turbo

Specifically, when AC is connected, this button cycles between Quiet -> Balanced -> Performance -> Turbo, and when AC is disconnected, it goes Eco -> Balanced.

I looked at this driver in mainline and found that such a feature already exists for Predator v4 Sense. It looks like my Nitro laptos uses the same WMI methods.

So I locally merged your version of this driver with the upsream one and additionaly patched it to support my laptop. And now it works, I can successfully switch between these modes!

I assume this new method will work for all recent Predator and Nitro laptops as well.

So here is the question. Do you plan to rebase your driver to mainline version? In that case, I can just wait and then create a PR with only the relevant changes. Or, I can create a PR to backport all the new features from upstream and include my patches for my laptop in that PR.

In any case, I plan to submit my patch for the mainline, but since this will be my first public patch for the Linux I have no idea when/if it will be applied.

**Comment by JafarAkhondali** [comment]
Dear @overcrook 
Thanks for your kind words.
I don't own the Acer laptop anymore thus it's hard for me to rebase without test. If you want to be a maintainer I can happily add you to the project. IMHO best thing to do for now is to resync this project with kernel mainline but also send fixes to kernel as well.
Some features can't be shipped to the kernel for now(e.g. RGB light control).
So creating a PR to backport everything should be the best option. We can even look for a method to apply them more cleanly so that we can rebase everything in a timely manner until everything can be shipped to the kernel.
I'm pretty sure you can apply it to kernel, just make sure to check the coding style requirements and use a CLI client send patches not GUI(disclaimer: I used GUI and corrupted indentations :D)

**Comment by overcrook** [comment]
OK, I'll prepare a PR soon, first I need to clean up the code a bit.
